extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -280,4 +203,112 @@ protected function parseImageData($data) {
 
 		return $images;
 	}
+
+	private function handlePostAction() {
+		$data = $this->wg->request->getValues();
+		if ( !empty($data) ) {
+			$images = $this->parseImageData($data);
+
+			if ( count($images) > 0 ) {
+				$this->helper->updateImageState( $images, $this->action );
+			}
+			$this->ts = null;
+		}
+	}
+
+	private function getOrderingMethod() {
+		if ($this->wg->user->isAllowed('imagereviewcontrols')) {
+			$order = (int)$this->getVal('sort', -1);
```

Comment:
maybe read preference, check is it different and set if it's different.
In theory: Set and Read can bring race condition.